### PR TITLE
Vim: enable sending multiple keystrokes from custom keybinding

### PR DESCRIPTION
**Pull request by avi-cenna**
Release Notes:

- Added `workspace::SendKeystrokes` to enable mapping from one key to a sequence of others ([#7033](https://github.com/zed-industries/zed/issues/7033)). 

Improves #7033. Big thank you to @ConradIrwin who did most of the heavy lifting on this one.

This PR allows the user to send multiple keystrokes via custom keybinding. For example, the following keybinding would go down four lines and then right four characters.

```json
[
  {
    "context": "Editor && VimControl && !VimWaiting && !menu",
    "bindings": {
      "g z": [
        "workspace::SendKeystrokes",
        "j j j j l l l l"
      ],
    }
  }
]
```

### CODE REVIEW

**Comment by ConradIrwin** [comment]
@avi-cenna Thanks for pushing this up!

#8019 fixes some of the issues we were seeing; but there are a number of cases we weren't thinking about on Friday that I'll add code for to this branch.

File: crates/workspace/src/workspace.rs [hunk 1]
```diff
@@ -1259,21 +1261,31 @@ impl Workspace {
...
-                cx.dispatch_event(PlatformInput::KeyDown(KeyDownEvent {
-                    keystroke: keystroke.clone(),
-                    is_held: false,
-                }));
-                cx.dispatch_event(PlatformInput::KeyUp(KeyUpEvent { keystroke }));
-            }
-        })
+        keystrokes.reverse();
+
+        self.dispatching_keystrokes
```
Comment (avi-cenna) [inline crates/workspace/src/workspace.rs:1271]: What's the purpose of this `dispatching_keystrokes` field? It looks like the parsed keystrokes are appended to this field, but I don't see where the field is being used outside of this.

**Comment by avi-cenna** [comment]
Thanks @ConradIrwin . I was looking over the changes and just had some questions out of curiosity.

File: crates/workspace/src/workspace.rs [hunk 2]
```diff
@@ -1259,21 +1261,31 @@ impl Workspace {
...
-        cx.window_context().defer(move |cx| {
-            for keystroke in keystrokes {
-                cx.dispatch_event(PlatformInput::KeyDown(KeyDownEvent {
-                    keystroke: keystroke.clone(),
-                    is_held: false,
-                }));
-                cx.dispatch_event(PlatformInput::KeyUp(KeyUpEvent { keystroke }));
-            }
-        })
+        keystrokes.reverse();
```
[the remark this answers is not in the document]
Comment (ConradIrwin) [reply crates/workspace/src/workspace.rs:1269]: Vectors in Rust are designed to be pushed/popped on the end; so this just makes the code a little more efficient. If you do `.remove(0)` to pull the first item out of the vector then it rewrites the entire vector, but `.pop()` can just change the length.

File: crates/workspace/src/workspace.rs (hunk 1 above)
Comment (ConradIrwin) [reply crates/workspace/src/workspace.rs:1271]: It allows you to include keys that themselves are bound to `SendKeystrokes`, so you can call this recursively.

In order to do that reliably we need to make sure that we append the new keystrokes to the existing list to be sure they'll be processed in order.

**Pull request merged**